### PR TITLE
Add handling of dependents and household units

### LINKED ISSUE #5: Add Italy dependent options

### LINKED ISSUE #26: Update modeling of individuals, households, etc.

### PR DESCRIPTION
**Pull request by anth-volk**
Fixes #5 
Fixes #26

### CODE REVIEW

File: policyengine_it/variables/household/person/is_dependent_between_18_and_21.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from policyengine_it.model_api import *
+
+
+class is_dependent_between_18_and_21(Variable):
```
Comment (MaxGhenis) [inline policyengine_it/variables/household/person/is_dependent_between_18_and_21.py]: We don't typically create variables with values hard coded. Are 18 and 21 policy parameters for some purpose?

Comment (anth-volk) [reply policyengine_it/variables/household/person/is_dependent_between_18_and_21.py]: They factor into Italian social programs - dependents aged 18 to 21 (and also 21 to 24) qualify for different benefits

Comment (anth-volk) [reply policyengine_it/variables/household/person/is_dependent_between_18_and_21.py]: Meeting with Pavel now, who's helping me to better understand how to structure these

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Handling of dependents and household units
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml]:
```suggestion
    - Handling of dependents and household units.
```

File: policyengine_it/variables/household/person/is_dependent.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from policyengine_it.model_api import *
+
+
+class is_dependant(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_it/variables/household/person/is_dependent.py]:
```suggestion
class is_dependent(Variable):
```

File: policyengine_it/variables/household/person/is_dependent.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    value_type = bool
+    entity = Person
+    label = "Is a dependant"
```
Comment (PavelMakarchuk) [inline policyengine_it/variables/household/person/is_dependent.py]:
```suggestion
    label = "Is a dependent"
```

File: policyengine_it/variables/household/person/is_dependent.py
Comment (PavelMakarchuk) [inline policyengine_it/variables/household/person/is_dependent.py:1]: Every variable that has a formula need a corresponding unit test, 

E.g. `count_dependents` doesnt need a unit test bc it is just an adds but `is_dependent` would need one 

Inputs:
every variable class that is used as inputs.

Here: `is_head` & `is_spouse`

Output:
class name 
here: `is_dependent`

We try to test every possible edge case

Comment (PavelMakarchuk) [reply policyengine_it/variables/household/person/is_dependent.py:1]: If we have time constraints, leave out

File: policyengine_it/variables/household/person/is_dependent_between_18_and_21.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    value_type = bool
+    entity = Person
+    label = "Is dependent between 18 and 21 years old"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        return (
+            person("age", period) >= 21
```
Comment (PavelMakarchuk) [inline policyengine_it/variables/household/person/is_dependent_between_18_and_21.py]: we would generally make those policy parameters unless the ages are defined by the Italian general legal code. We can make separate parameters for the '18' and '21' especially since there is also a limitation for children from 21 to 24

Since we define the child to be under 18 as well we might consider using the same parameter as in that file

For the naming if this is just used for a specific credit then we would name this 
`older_child_for_xyz_credit`  and store the parameters in the respective credit parameter folder

File: policyengine_it/variables/household/person/is_dependent_between_21_and_24.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    value_type = bool
+    entity = Person
+    label = "Is between 21 and 24 years old"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        return (
+            person("age", period) >= 21
+            and person("age", period) < 24
+            and person("is_dependent", period) == True
```
Comment (PavelMakarchuk) [inline policyengine_it/variables/household/person/is_dependent_between_21_and_24.py]:
```suggestion
           age = person("age", period)
            meets_age_reguirement = p.younger =< age < p.older
            dependent = person("is_dependent", period)
           return meets_age_reguirement & dependent
```
We try to define each "action" in separate variables e.g. above

Comment (PavelMakarchuk) [reply policyengine_it/variables/household/person/is_dependent_between_21_and_24.py]: p stands for the parameter path: 
What we commonly use is:
```
p = parameters(period).person.age
```

File: policyengine_it/variables/household/person/is_under_1.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+from policyengine_it.model_api import *
+
+
+class is_dependent_under_1(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_it/variables/household/person/is_under_1.py]: same as above

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_it/variables/household/person/is_dependent_between_21_and_24.py
Comment (PavelMakarchuk) [inline policyengine_it/variables/household/person/is_dependent_between_21_and_24.py:1]: we can make a folder called `dependent` with all dependent related vars in `person`

**Comment by anth-volk** [comment]
Tests are expected to fail; no params exist yet
